tests/unit/actions/settings/test_settings.py

```diff
@@ -0,0 +1,151 @@
+"""Unit tests for the ``settings`` action."""
+
+from dataclasses import asdict
+from dataclasses import dataclass
+from typing import Optional
+
+import pytest
+
+from ansible_navigator.actions.settings import CONTENT_HEADING_DEFAULT
+from ansible_navigator.actions.settings import CONTENT_HEADING_NOT_DEFAULT
+from ansible_navigator.actions.settings import color_menu
+from ansible_navigator.actions.settings import content_heading
+from ansible_navigator.actions.settings import filter_content_keys
+from ansible_navigator.ui_framework import Color
+from ansible_navigator.ui_framework import CursesLinePart
+from ansible_navigator.ui_framework import Decoration
+
+
+@dataclass
+class ColorMenuTestEntry:
+    """A test for menu coloring."""
+
+    color: int
+    """The color for menu entry"""
+    comment: str
+    """Describe the test"""
+    decoration: int
+    """The test decoration"""
+    is_default: bool
+    """Is the current value equal to the default"""
+
+    def __str__(self):
+        """Provide a string representation.
+
+        :returns: The string representation of self
+        """
+        return self.comment
+
+
+ColorMenuTestEntries = (
+    ColorMenuTestEntry(
+        comment="default/green",
+        color=Color.GREEN,
+        decoration=Decoration.NORMAL,
+        is_default=True,
+    ),
+    ColorMenuTestEntry(
+        comment="not default/yellow",
+        color=Color.YELLOW,
+        decoration=Decoration.NORMAL,
+        is_default=False,
+    ),
+)
+
+
+@pytest.mark.parametrize(argnames="data", argvalues=ColorMenuTestEntries, ids=str)
+def test_color_menu(data: ColorMenuTestEntry):
+    """Test color menu for a val set to the default.
+
+    :param data: A test entry
+    """
+    entry = {"is_default": data.is_default}
+    assert color_menu(0, "", entry) == (data.color, data.decoration)
+
+
+@dataclass
+class ContentHeadingEntry:
+    """A test for content headings."""
+
+    color: int
+    """The color for menu entry"""
+    comment: str
+    """Describes the test"""
+    current_value: str
+    """The current value"""
+    default: str
+    """The default value"""
+    is_default: Optional[bool] = None
+    """Indicate if the current is equal to the default."""
+    name: str = "test settings entry"
+    """The name of the settings entry"""
+
+    def __post_init__(self):
+        """Set is_default after initialization."""
+        self.is_default = self.current_value == self.default
+
+    @property
+    def heading(self):
+        """Create the expected heading for this content.
+
+        :returns: The expected heading
+        """
+        if self.is_default:
+            heading = CONTENT_HEADING_DEFAULT
+        else:
+            heading = CONTENT_HEADING_NOT_DEFAULT
+        return heading.format(**asdict(self))
```

Comment:
FYI `str.format_map()` exists
```suggestion
        return heading.format_map(asdict(self))
```

Reply:
Does this have additional benefits besides directly accessing the dict values?

Reply:
Documentation says: _format_map() function is useful when we want to use a dict subclass for mapping purposes._ It still works, but since we're just using a dict and not a subclass of dict, should we change it?

Reply:
If I saw format_map, I would wonder why a simple ** wasn't used and probably go back to the source object to see if something funny was happening.

That said, your call @alisonlhart

Reply:
I think the only benefit is that .format_map was introduced more recently than .format and is specifically for mapping, but it seems unnecessary in this case. I say we leave it as .format(**mapping).

Reply:
I suspected that `**` may be slower because it's decomposing the dict before converting it into kwargs and calling the method. OTOH with `str.format_map()` that dict is just passed in by reference as is. And it seems I may be onto something:
```python
In [1]: %timeit ''.format(**locals())
308 ns ± 25.4 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)

In [2]: %timeit ''.format_map(locals())
167 ns ± 6.77 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)
```